packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -191,14 +191,20 @@ export class SCWSigner implements Signer {
         this.storage.storeObject(ACCOUNTS_KEY, accounts);
 
         // TODO: in future PR update state to support multiple accounts
-        const account = response.accounts[0];
-        const capabilities = account.capabilities;
-        if (capabilities && capabilities.addAddress) {
+        const account = response.accounts.at(0);
+        const capabilities = account?.capabilities;
+        if (capabilities?.addAddress) {
           assertSubAccountInfo(capabilities.addAddress);
           subaccounts.setState({
             account: capabilities.addAddress,
           });
         }
+        if (capabilities?.getAppAccounts) {
+          assertSubAccountInfo(capabilities.getAppAccounts[0]);
+          subaccounts.setState({
+            account: capabilities.getAppAccounts[0],
+          });
+        }
         this.callback?.('accountsChanged', [
```

Comment:
will state be overwritten if both capabilities are set?

Reply:
ooo good callout. I think we need to handle this a bit differently